src/Microsoft.Azure.WebJobs.Host/Scale/DefaultHostProcessMonitor.cs

```diff
@@ -66,7 +69,9 @@ public void UnregisterChildProcess(Process process)
             ProcessMonitor monitor = null;
             lock (_syncLock)
             {
-                monitor = _childProcessMonitors.SingleOrDefault(p => p.Process == process);
+                // try to locate the monitor for the specified process
+                // and remove it if found
+                monitor = _childProcessMonitors.SingleOrDefault(p => p.Process.Inner == process);
```

Comment:
Having to access Inner in product code kind of breaks the abstraction. I get this interface was introduced for testing purposes, but once you incur the cost of introducing the interface, I'd like to try to stick to it wherever possible. Could we do this lookup by ID instead?

Reply:
Yes I realize that and I struggled here. The issue is that when the process is closed/disposed it's property accessors throw InvalidOperation exceptions. So I can't go by ID unfortunately.

Reply:
I've resolved this issue by removing the Inner property and instead overriding Equals, allowing a wrapper to compare (and be equal to) the process instance it wraps.